crates/core/src/metadata/manager.rs

```diff
@@ -342,6 +393,11 @@ impl MetadataManager {
             return Ok(());
         }
 
+        // avoid syncing metadata we don't care about
+        if !self.metadata_mask.contains(metadata_kind) {
+            return Ok(());
```

Comment:
This is just a sketch, maybe there's a better way to fuse updates by metadata kind.